src/player.cpp

```diff
@@ -12295,9 +12295,12 @@ std::vector<const item *> player::all_items_with_flag( const std::string &flag )
     } );
 }
 
-bool player::has_item_with_flag( const std::string &flag ) const
+bool player::has_item_with_flag( const std::string &flag, bool need_charges ) const
 {
-    return has_item_with( [&flag]( const item & it ) {
+    return has_item_with( [&flag, &need_charges]( const item & it ) {
+        if( it.is_tool() && need_charges ) {
+            return it.has_flag( flag ) && it.type->tool->max_charges ? it.charges > 0 : it.has_flag( flag );
```

Comment:
```suggestion
            return it.has_flag( flag ) && it.type->tool->max_charges ? it.charges > 0 : true;
```

Reply:
I guess this change caused this bug: #28619.